### PR TITLE
feature 10/12 vote version 1

### PR DESCRIPTION
**Pull request by bimu448**
feature 10/12 version 1

### CODE REVIEW

**Comment by KevinHChen** [comment]
Would you please specify the changes in the description?
Also, it's my bad, I explained to you wrongly yesterday. Instead, you should commit the latest package-lock file as well because we are using `npm ci.`

**Comment by MarkZhuVUW** [comment]
I will commit to this pr and help fix conflict and merge the changes with my branch

File: frontend/package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "@mui/icons-material": "^5.15.14",
     "@mui/material": "^5.15.13",
     "@mui/system": "^5.15.14",
+    "antd": "^5.16.4",
```
Comment (MarkZhuVUW) [inline frontend/package.json:20]: I think it is better that we remove this and use mui components to ensure consistent ui

File: frontend/src/components/Voting/style.css [hunk 2]
```diff
@@ -0,0 +1,116 @@
+@import url("https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700&display=swap");
+
+body,
```
Comment (MarkZhuVUW) [inline frontend/src/components/Voting/style.css:3]: css could break mui component styling especially if you override body, h2 etc

Better if we can use mui built in way of tweaking css

File: backend/controllers/voteControllers.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+      .status(200)
+      .json({ code: 20000, data: null, message: "update successfully" });
+  } catch (error) {
+    return res
+      .status(500)
+      .json({ code: 50000, data: null, error: error.message });
+  }
+});
+
+router.post("/endVote", async (req, res) => {
```
Comment (MarkZhuVUW) [inline backend/controllers/voteControllers.js:139]: ending a vote should be a put api

ending a vote is the same as updating a vote

File: backend/controllers/voteControllers.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+});
+
+/**
+ * @swagger
+ * /api/votes/{voteId}:
+ *   put:
+ *     tags:
+ *     - Vote Controller
+ *     summary: Update a vote
+ *     parameters:
```
Comment (MarkZhuVUW) [inline backend/controllers/voteControllers.js:91]: apis do not follow RESTful design and swagger ui docs are not working

**Review by MarkZhuVUW — CHANGES_REQUESTED** [review]

**Comment by KevinHChen** [comment]
Known issues are all fixed.
1. Another CSS library is removed.
2. First version of voting page works now

There are still some other bugs. I'll create a ticket to track them.

**Review by KevinHChen — APPROVED** [review]

File: frontend/src/components/Voting/style.css (hunk 2 above)
Comment (KevinHChen) [reply frontend/src/components/Voting/style.css:3]: This shall be outdate

**Pull request merged**